pkg/storage/tsdb/config.go

```diff
@@ -264,6 +264,10 @@ type TSDBConfig struct {
 
 	// HeadCompactionIntervalJitterEnabled is enabled by default, but allows to disable it in tests.
 	HeadCompactionIntervalJitterEnabled bool `yaml:"-"`
+
+	// TimelyHeadCompaction allows head compaction to happen when min block range can no longer be appended,
+	// without requiring 1.5x the chunk range worth of data in the head.
+	TimelyHeadCompaction bool `yaml:"timely_head_compaction" category:"experimental"`
```

Comment:
[nit] Do you mind adding a `Enabled` at the end (both the field, YAML and CLI flag)?